include/sway/tree/container.h

```diff
@@ -87,54 +87,77 @@ struct sway_container {
 	} events;
 };
 
-// TODO only one container create function and pass the type?
-struct sway_container *container_output_create(
-		struct sway_output *sway_output);
+// TODO make private and use the container-specific create functions
+struct sway_container *container_create(enum sway_container_type type);
 
-struct sway_container *container_workspace_create(
-		struct sway_container *output, const char *name);
+/**
+ * Create a new container container. A container container can be a a child of
```

Comment:
We should consider renaming containers to nodes

Reply:
The word `container` is used in the documentation and some commands. Renaming it will either break people's configs or split the name for it between the code and the user docs which would be confusing.

Reply:
I think it would be wise to rename it internally, even if we don't update the docs.

Reply:
Ok let's do it in another pr.